### PR TITLE
(POC) MongoDB functions, endpoints and react hooks.

### LINKED ISSUE #46: Create API endpoints and MongoDB functions.
*Create MongoDB functions for CRUD.
*Create routes and endpoints.

### PR DESCRIPTION
**Pull request by jaanikaukonen**
fixes #46 
### Backend
* Moved MongoDB connection in `config/db.ts`.
* Made app to use created routes.
* Controllers:
   * `channelController.ts`
   * `deviceController.ts`
* Routes:
   * `channelRoutes.ts`
   * `deviceRoute.ts`
   
### Frontend
* Added `useDevices` to `apiHooks.ts`.
* Types:
    * `deviceHistoryType.ts`
    * `deviceSensorType.ts`
    * `deviceType.ts`
    * `sensorDataType.ts`

### CODE REVIEW

File: front-end/src/hooks/ApiHooks.ts [hunk 1]
```diff
@@ -29,4 +29,17 @@ const useWebSocket = () => {
...
+    const fectDevices = async () => {
+      const response = await fetch(apiUrl)
+      const data = await response.json()
+      setDevices(data)
+    }
+
+    fectDevices()
+  }, [])
+  return devices
+}
```
Comment (NikoLindborg) [inline front-end/src/hooks/ApiHooks.ts:45]: modify it to fit the hooks rules, for example the larger function "useDevices" which can then return smaller device related states which are then destructured

File: front-end/src/App.tsx [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-import DeviceListComponent from './components/DeviceListComponent/DeviceListComponent'
-import ChannelIcon from './assets/icons/channel_icon.svg'
-import StatusRedIcon from './assets/icons/status_red.svg'
-import useWebSocket from './hooks/ApiHooks'
+import {useWebSocket, getDevices} from './hooks/ApiHooks'
 import BaseView from './views/base/BaseView'
 
 const App: React.FC = () => {
   const {data} = useWebSocket()
+  const devices = getDevices()
```
Comment (NikoLindborg) [inline front-end/src/App.tsx:8]:
```suggestion
  const {devices} = getDevices()
```

File: front-end/src/hooks/ApiHooks.ts (hunk 1 above)
Comment (NikoLindborg) [inline front-end/src/hooks/ApiHooks.ts:45]: Modify it to fit the reacts hook rules. We can later on add more device related api calls into the same hook, and then return them, as needed, in the destructured return statement. This also requires you to change the function call in the app.ts file
```suggestion
export const useDevices = async () => {
  const [devices, setDevices] = useState([])

  useEffect(() => {
       fetchDevices ()
  }, [])
  const fetchDevices = async () => {
      const response = await fetch(apiUrl)
      const data = await response.json()
      setDevices(data)
    }
  return {devices}
}
```

Comment (jaanikaukonen) [reply front-end/src/hooks/ApiHooks.ts]: Did as suggested.

**Review by NikoLindborg — APPROVED** [review]
Looks good with the adjustments

**Review by aleksikyto — APPROVED** [review]
Looking good.

**Pull request merged**